lib/Dialect/HW/Transforms/FlattenIO.cpp

```diff
@@ -349,6 +349,15 @@ static LogicalResult flattenOpsOfType(ModuleOp module, bool recursive) {
       });
     });
 
+    DenseMap<Operation *, ArrayAttr> oldArgNames, oldResNames, oldArgLocs,
+        oldResLocs;
+    for (auto op : module.getOps<T>()) {
+      oldArgNames[op] = op->template getAttrOfType<ArrayAttr>("argNames");
```

Comment:
Aren't argNames now part of the ModuleType not separate attributes?

Reply:
Or is this for backwards compat?

Reply:
This hasn't moved anything over.  This and lots of other code is randomly poking attributes by name and it will all break.  But that's tomorrow's problem.  Today is just decoupling from FunctionOpInterface.